### PR TITLE
Postgres dialect parse password with spaces (#775)

### LINKED ISSUE #775: postgres dialect panics if password contains a space
With the following code (the password should be escaped but isn't):

```go
cntStr := "postgres://badpass:test space@localhost:5432/grafana?sslmode=disable"
engine, err := xorm.NewEngine("postgres", cntStr)
```

The [parseOpts](https://github.com/go-xorm/xorm/blob/master/dialect_postgres.go#L1178-L1182) method panics instead of returning an error.

Would you accept a PR that does the following:

Removes the [errorf function](https://github.com/go-xorm/xorm/blob/master/dialect_postgres.go#L1120) that is used in the parseOpts function and that returns an error if the parsing fails.

### PR DESCRIPTION
**Pull request by daniellee**
Fixes #775 This fixes two problems - the postgres dialect can parse a password with spaces. If an error occurs the Parse func returns an error instead of panicking.

There are some problems still however:

- The Parse function cannot handle symbols like `#` in a password but we handle that in Grafana (where I found this bug) by escaping everything before creating the connection string. 

The code seems to be based on code from pq from 5 year ago. The current version of pq does a lot more checks when parsing connection strings to handle all these cases - see this [PR](https://github.com/lib/pq/pull/133) that fixed the panic problem in pq.

### CODE REVIEW

**Comment by daniellee** [comment]
Working on fixing the coverage tests - not sure exactly why are failing yet.

**Comment by daniellee** [comment]
Aha, was a little too aggressive with removing code. Needs to be able to parse opts and not just urls.

File: dialect_postgres.go [hunk 1]
```diff
@@ -1179,31 +1146,40 @@ func parseOpts(name string, o values) {
...
+
+	return nil
 }
 
 func (p *pqDriver) Parse(driverName, dataSourceName string) (*core.Uri, error) {
 	db := &core.Uri{DbType: core.POSTGRES}
-	o := make(values)
-	var err error
+	var (
+		dbName string
```
Comment (lunny) [inline dialect_postgres.go]: It seems local variable `dbName` is unnecessary.

Comment (daniellee) [reply dialect_postgres.go]: You mean it should be inlined and db.DbName should be set directly on line 1165 and 1176?

Comment (daniellee) [reply dialect_postgres.go]: Why is that a bug? It is not an unused variable.

Comment (lunny) [reply dialect_postgres.go]: @daniellee yes, I think so.
This PR will fix a bug issue so that I labeled it `bug`.

**Comment by daniellee** [comment]
@lunny Changes made. Is that what you were thinking?

**Comment by lunny** [comment]
@daniellee yes and merged.

**Pull request merged**